src/wallet/wallet.cpp

```diff
@@ -4209,19 +4205,19 @@ static const std::string OUTPUT_TYPE_STRING_LEGACY = "legacy";
 static const std::string OUTPUT_TYPE_STRING_P2SH_SEGWIT = "p2sh-segwit";
 static const std::string OUTPUT_TYPE_STRING_BECH32 = "bech32";
 
-OutputType ParseOutputType(const std::string& type, OutputType default_type)
+bool ParseOutputType(const std::string& type, OutputType& output_type)
 {
-    if (type.empty()) {
-        return default_type;
-    } else if (type == OUTPUT_TYPE_STRING_LEGACY) {
-        return OutputType::LEGACY;
+    if (type == OUTPUT_TYPE_STRING_LEGACY) {
```

Comment:
Perhaps it would be good to support explicitly selecting the auto behaviour (with a "auto" string, only available for change?)

Otherwise it's hard to override a config file setting that set it to something else e.g.

Reply:
FYI https://github.com/bitcoin/bitcoin/pull/12729#issuecomment-374779824:

> > How about explicit setting like -changetype=auto
> 
> Good idea for another PR, but for now I'm just trying to do a code cleanup.

Reply:
Oh, I missed that. Sounds reasonable.